mutation-testing/pr-differences/action.yml

```
@@ -0,0 +1,173 @@
+name: PR Differences Mutants
+
+inputs:
+  shard:
+    description: "The number of the shard to run (`-1` if ran without shards)"
+    required: false
+    default: -1
+  package-dimension:
+    description: "The dimension of the package. `big` for `stacks-node` and `stackslib`, `small` for others"
+    required: true
+
+runs:
+  using: "composite"
+
+  steps:
+    # Cleanup Runner
+    - name: Cleanup Runner
+      id: runner_cleanup
+      uses: stacks-network/actions/cleanup/disk@main
+
+    # Checkout the stacks-core code
+    - name: Checkout stacks-core repo
+      id: git_checkout_stacks_core
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        fetch-depth: 0
+
+    - name: Relative diff
+      shell: bash
+      run: |
+        git diff origin/${{ github.base_ref }}.. > git.diff
+
+    # Checkout the actions code
+    - name: Checkout actions repo
+      id: git_checkout_actions
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        repository: stacks-network/actions
+        ref: feat/mutation-testing
+        path: ./actions-repo
+
+    - name: Copy git diff
+      shell: bash
+      run: |
+        if [ ! -s git.diff ]; then
+          echo "Diff file missing!"
```

if the file is not found/empty - we should exit 
```suggestion
          echo "Diff file missing!"
          exit 1
```